gadget/gadget.go

```diff
@@ -323,6 +323,17 @@ func SetEnclosingVolumeInStructs(vv map[string]*Volume) {
 	}
 }
 
+// ResetEnclosingVolumeInStructs removes all recursive pointers
+// added by SetEnclosingVolumeInStructs. This is useful
+// to be able to serialize Volume.
+func ResetEnclosingVolumeInStructs(vv map[string]*Volume) {
```

Comment:
Maybe just a very simple unit test for this?

Reply:
Because this has to be reverted when ubuntu-image has stopped using it, I do not care if anyone breaks it.

Maybe I should make it more clear in the comment that this will be eventually removed.